superset/jinja_context.py

```diff
@@ -722,3 +723,69 @@ def dataset_macro(
     sqla_query = dataset.get_query_str_extended(query_obj, mutate=False)
     sql = sqla_query.sql
     return f"(\n{sql}\n) AS dataset_{dataset_id}"
+
+
+def get_dataset_id_from_context(metric_key: str) -> int:
+    """
+    Retrives the Dataset ID from the request context.
+
+    :param metric_key: the metric key.
+    :returns: the dataset ID.
+    """
+    # pylint: disable=import-outside-toplevel
+    from superset.daos.chart import ChartDAO
+    from superset.views.utils import get_form_data
```

Comment:
do we really need to import these outside of the toplevel? even get_form_data?

Reply:
@dpgaspar I was following the pattern from other methods in this file -- happy to move them to the top. Going to do that in my next commit.

Reply:
@dpgaspar I tried moving these local imports to the top but they result in circular import errors (even the `get_form_data`. I was only able to move the test imports to the top.